packages/manager/apps/pci-private-registry/src/pages/IAM/ManageIAM.page.tsx

```diff
@@ -0,0 +1,107 @@
+import { useCallback, useContext } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { Translation } from 'react-i18next';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { IAMModal } from './IAMModal/IAMModal';
+import { useGetCachedRegistry } from '@/api/hooks/useRegistry';
+import { useToggleIAMAuthentication } from '@/api/hooks/useIAMAuthentication';
+import { TRegistryAction, TRegistryActionToggle } from './constants';
+
+const ManageIAM = () => {
+  const navigate = useNavigate();
+  const { shell } = useContext(ShellContext);
+  const { addError, addSuccess } = useNotifications();
+  const { projectId, registryId } = useParams();
+
+  if (!projectId || !registryId) return null;
+
+  const registry = useGetCachedRegistry(projectId, registryId);
```

Comment:
This works, but regarding the rule of hooks, you cannot invoke a hook conditionally, React hooks must be called in the exact same order in every component render.
So, you'll need to find another way to orchestrate data flow in the page :)